app/controllers/admin/categories_controller.rb

```diff
@@ -24,13 +24,21 @@ def destroy
   private
 
   def new_or_edit
+    if params[:action] == "new"
+      @category = Category.new
+    elsif params[:action] == "edit" && params[:id]
+      @category = Category.find(params[:id])
+      @category.attributes = params[:category]
+    elsif params[:action] == "edit" && !params[:id]
```

Comment:
The first "if" statement is for when you first navigate to the page by clicking the "Categories" link (admin/categories/new). The first "elsif" is for when you navigate to to "/admin/categories/edit/:id" by clicking the "edit" link for a category. The second "elsif" is to handle the post request (which uses the "edit" action) when the category is new (which is why there wouldn't be an id in the params).